src/Lib.P2P/SwarmService.cs

```diff
@@ -0,0 +1,1132 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.NetworkInformation;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+using Common.Logging;
+using Lib.P2P.Cryptography;
+using Lib.P2P.Protocols;
+using Lib.P2P.Routing;
+using Lib.P2P.SecureCommunication;
+using Lib.P2P.Transports;
+using MultiFormats;
+using Nito.AsyncEx;
+
+namespace Lib.P2P
+{
+    /// <summary>
+    ///   Manages communication with other peers.
+    /// </summary>
+    public class SwarmService : ISwarmService
+    {
+        private static ILog _log = LogManager.GetLogger(typeof(SwarmService));
+
+        /// <summary>
+        ///   The time to wait for a low level connection to be established.
+        /// </summary>
+        /// <value>
+        ///   Defaults to 30 seconds.
+        /// </value>
+        public TimeSpan TransportConnectionTimeout = TimeSpan.FromSeconds(30);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make this field 'private' and encapsulate it in a 'public' property.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=5051852)